airbyte-integrations/connectors/source-stripe/source_stripe/source.py

```diff
@@ -513,12 +518,25 @@ def streams(self, config: MutableMapping[str, Any]) -> List[Stream]:
                 },
                 **args,
             ),
-            StripeLazySubStream(
+            UpdatedCursorIncrementalStripeLazySubStream(
                 name="invoice_line_items",
                 path=lambda self, stream_slice, *args, **kwargs: f"invoices/{stream_slice['parent']['id']}/lines",
                 parent=invoices,
+                event_types=[
+                    "invoice.created",
+                    "invoice.deleted",
+                    "invoice.updated",
+                    # the event type = "invoice.upcoming" doesn't contain the `primary_key = `id` field,
+                    # thus isn't used, see the doc: https://docs.stripe.com/api/invoices/object#invoice_object-id
+                    # reference issue: https://github.com/airbytehq/oncall/issues/5560
+                ],
                 sub_items_attr="lines",
-                slice_data_retriever=lambda record, stream_slice: {"invoice_id": stream_slice["parent"]["id"], **record},
+                slice_data_retriever=lambda record, stream_slice: {
+                    "invoice_id": stream_slice["parent"]["id"],
```

Comment:
I don't know why invoice_id is in this stream(it was there before this change), but the column `invoice` contains the id of the invoice

Reply:
As this is already a field in the current stream, we can't remove that without a breaking change so I would keep it for now. Can we add a comment to say that this should be removed eventually as the information is already present in the field `invoice`?

Reply:
Since this existed prior to my change, I'll leave this for you guys to comment/change as you see fit.